package/src/components/Poll/CreatePollContent.tsx

```diff
@@ -1,214 +1,126 @@
   const { t } = useTranslationContext();
-  const [pollTitle, setPollTitle] = useState('');
-  const [pollOptions, setPollOptions] = useState<PollOptionData[]>([{ text: '' }]);
-  const [multipleAnswersAllowed, setMultipleAnswersAllowed] = useState(false);
-  const [maxVotesPerPerson, setMaxVotesPerPerson] = useState('');
-  const [maxVotesPerPersonEnabled, setMaxVotesPerPersonEnabled] = useState(false);
-  const [isAnonymous, setIsAnonymous] = useState(false);
-  const [optionSuggestionsAllowed, setOptionSuggestionsAllowed] = useState(false);
-  const [commentsAllowed, setCommentsAllowed] = useState(false);
-  const [duplicates, setDuplicates] = useState<string[]>([]);
 
-  const { closePollCreationDialog, createAndSendPoll } = useCreatePollContentContext();
+  const messageComposer = useMessageComposer();
+  const { pollComposer } = messageComposer;
+  const { allow_answers, allow_user_suggested_options, options, voting_visibility } = useStateStore(
+    pollComposer.state,
+    pollComposerStateSelector,
+  );
+
+  const { createPollOptionHeight, closePollCreationDialog, createAndSendPoll } =
+    useCreatePollContentContext();
 
   // positions and index lookup map
   // TODO: Please rethink the structure of this, bidirectional data flow is not great
   const currentOptionPositions = useSharedValue<CurrentOptionPositionsCache>({
-    inverseIndexCache: { 0: 0 },
-    positionCache: { 0: { updatedIndex: 0, updatedTop: 0 } },
+    inverseIndexCache: {},
+    positionCache: {},
   });
 
   const {
     theme: {
-      colors: { accent_error, bg_user, black, white },
+      colors: { bg_user, black, white },
       poll: {
-        createContent: {
-          addComment,
-          anonymousPoll,
-          headerContainer,
-          maxVotes,
-          multipleAnswers,
-          name,
-          scrollView,
-          sendButton,
-          suggestOption,
-        },
+        createContent: { addComment, anonymousPoll, scrollView, suggestOption },
       },
     },
   } = useTheme();
 
   useEffect(() => {
-    const seenTexts = new Set<string>();
-    const duplicateTexts = new Set<string>();
-    for (const option of pollOptions) {
-      const { text } = option;
-      if (seenTexts.has(text)) {
-        duplicateTexts.add(text);
-      }
-      if (text.length > 0) {
-        seenTexts.add(text);
-      }
-    }
+    if (!createPollOptionHeight) return;
+    options.forEach((option, index) => {
+      currentOptionPositions.value = {
+        ...currentOptionPositions.value,
+        inverseIndexCache: {
+          ...currentOptionPositions.value.inverseIndexCache,
+          [index]: option.id,
+        },
+        positionCache: {
+          ...currentOptionPositions.value.positionCache,
+          [option.id]: {
+            updatedIndex: index,
+            updatedTop: index * createPollOptionHeight,
+          },
+        },
+      };
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Please fix the dependencies here rather than silencing the rule

Reply:
Well we don't want the effect to run with every change of options and only on options length that is why I added it and it was needed.

Reply:
Why exactly do we not want that ? We very much do.

Reply:
No, I mean, I don't want the logic inside to rerun every time there is a change in options(including the ordering), but only do it when the length changes, cause here that's what is needed. Functionality wise no difference even if I add `options` but I took it as an optimization.

Reply:
It should pretty much produce the same thing, right ? Meaning, there shouldn't be any flickering animation wise. 

The point is, we can at most have 100 options. The optimization itself saves probably something in the order of picoseconds, while we break the hook dependency (for example, if we tomorrow decide that reordering suddenly depends on the content of each option or if we decide to provide integrators with a custom ordering function this will be wrong). As a matter of fact, nothing's stopping integrators from writing their own middleware for poll option creation that might depend on something else.

I would keep the dependency stable here.

Reply:
If we don't care about the re calculations, I don't mind removing the line

Reply:
I tested it on the device and it works fine. Pushed the changes.